src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -803,77 +719,21 @@ private void LoadAssemblies(IEnumerable<string> assemblies)
-            foreach (var type in assembly.GetTypes())
-            {
-                if (s_sourceCache.ContainsKey(type.FullName))
-                {
-                    if (s_sourceCache[type.FullName] != sourceCode.GetHashCode())
-                    {
-                        ErrorRecord errorRecord = new ErrorRecord(
-                                new Exception(
-                                    String.Format(AddTypeStrings.TypeAlreadyExists, type.FullName)),
-                                "TYPE_ALREADY_EXISTS",
-                                ErrorCategory.InvalidOperation,
-                                type.FullName);
-
-                        ThrowTerminatingError(errorRecord);
-                        return;
-                    }
-                }
-                else
-                {
-                    s_sourceCache[type.FullName] = sourceCode.GetHashCode();
-                }
-            }
-        }
-
-        // We now ship the NetCoreApp2.0 reference assemblies with PowerShell Core, so that Add-Type can work
-        // in a predictable way and won't be broken when we move to newer version of .NET Core.
-        // The NetCoreApp2.0 reference assemblies are located at '$PSHOME\ref'.
-        private static string s_netcoreAppRefFolder = PathType.Combine(PathType.GetDirectoryName(typeof(PSObject).Assembly.Location), "ref");
-        private static string s_frameworkFolder = PathType.GetDirectoryName(typeof(object).Assembly.Location);
-
-        // These assemblies are always automatically added to ReferencedAssemblies.
-        private static Lazy<PortableExecutableReference[]> s_autoReferencedAssemblies = new Lazy<PortableExecutableReference[]>(InitAutoIncludedRefAssemblies);
-
-        // A HashSet of assembly names to be ignored if they are specified in '-ReferencedAssemblies'
-        private static Lazy<HashSet<string>> s_refAssemblyNamesToIgnore = new Lazy<HashSet<string>>(InitRefAssemblyNamesToIgnore);
-
-        // These assemblies are used, when ReferencedAssemblies parameter is not specified.
-        private static Lazy<PortableExecutableReference[]> s_defaultAssemblies = new Lazy<PortableExecutableReference[]>(InitDefaultRefAssemblies);
-
-        private bool InMemory { get { return String.IsNullOrEmpty(outputAssembly); } }
-
         /// <summary>
         /// Initialize the list of reference assemblies that will be used when '-ReferencedAssemblies' is not specified.
         /// </summary>
-        private static PortableExecutableReference[] InitDefaultRefAssemblies()
+        private static IEnumerable<PortableExecutableReference> InitDefaultRefAssemblies()
         {
             // netcoreapp2.0 currently comes with 137 reference assemblies (maybe more in future), so we use a capacity of '150'.
             var defaultRefAssemblies = new List<PortableExecutableReference>(150);
+            Diagnostics.Assert(defaultRefAssemblies.Count < 150, "Size of defaultRefAssemblies is small.");
```

Comment:
Now we reach this threshold. We can exceed it after the transition to .Net Core 2.1. That's why I made this reminder.

Reply:
There are two problems here:
1. The assert at this location doesn't do anything useful because when the list is initially created, its count will be 0.
2. I don't know how many people uses debug mode binaries. It could happen that this assert fails without us knowing. This happened before to an assert in `NativeComamndParameterBinder`.

Reply:
Clear. Removed.